uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/CRM/Tasksassignments/BAO/AssignmentTest.php

```diff
@@ -17,42 +17,57 @@ class CRM_Tasksassignments_BAO_AssignmentTest extends \PHPUnit_Framework_TestCas
 
   private $open_status_id;
 
-  private $closed_status_id;
-
   public function setUpHeadless() {
     return \Civi\Test::headless()
       ->installMe(__DIR__)
       ->apply();
   }
 
   public function setUp() {
-    parent::setUp();
-    // Create Case Type
+    // Create Contact
     $params = ['first_name' => 'micky', 'last_name' => 'mouse'];
     $this->contact_id = $this->createContact($params);
-    // Create Contact
+    // Create Case Type
     $params = ['name' => 'test_case_type', 'title' => 'Test Case Type', 'is_active' => 1];
     $this->case_type_id = $this->createCaseType($params);
     // Create Case Status option values
     $params = ['name' => 'Opened', 'grouping' => 'Opened'];
     $this->open_status_id = $this->createOptionValue($params, 'case_status');
   }
 
-  public function tearDown() {
-    parent::tearDown();
+  public function testRetrieveCasesWithoutExcludeCaseIds() {
+    $params = [
+      'creator_id' => $this->contact_id,
+      'case_type_id' => $this->case_type_id,
+      'subject' => 'test case',
+      'status_id' => $this->open_status_id,
+    ];
+    $testCaseID1 = $this->createCase($params);
+
+    $params = [
+      'creator_id' => $this->contact_id,
+      'case_type_id' => $this->case_type_id,
+      'subject' => 'test case2',
+      'status_id' => $this->open_status_id,
+      'end_date' => date('Ymd')
+    ];
+    $testCaseID2 = $this->createCase($params);
+
+    $cases = CRM_Tasksassignments_BAO_Assignment::retrieveCases();
+
+    $this->assertCount(2, $cases);
+    $this->assertTrue(!empty($cases[$testCaseID1]));
```

Comment:
You can use the assertNotEmpty here `$this->assertNotEmpty($cases[$testCaseID1]);`

Even though this new assertion improves the test, I'm afraid this is still not good enough. If retrieveCases returns an array like
```
[
 1 => [1],
 2 => [2]
]
```

the test will still pass. Here's a suggestion of how to improve this test (also applies for the others):
```php
 $casesFixtures = [
  ['subject' => 'test case', 'end_date' => null],
  ['subject' => 'test case2', 'end_date' => date('Ymd')]
];

$defaultParams = [
  'creator_id' => $this->contact_id,
  'case_type_id' => $this->case_type_id,
  'status_id' => $this->open_status_id,
];

$createdCases = [];
foreach($casesFixtures as $caseFixture) {
  $params = array_merge($defaultParams, $casesFixtures);
  $caseID = $this->createCase($params);
  $createdCases[$caseID] = $params;
}

$cases = CRM_Tasksassignments_BAO_Assignment::retrieveCases();
$this->assertCount(count($createdCases), $cases);
foreach($cases as $caseID => $caseValues) {
  $this->assertNotEmpty($createdCases[$caseID]);
  $this->assertEquals($createdCases[$caseID]['subject'], $caseValues['case_subject']);
  $this->assertEquals($this->contact_id, $caseValues['contact_id']);
  $this->assertEquals($this->open_status_id, $caseValues['status_id']);
  // any other assertion that you might find relevant
}
```